packages/flutter_tools/lib/src/commands/packages.dart

```diff
@@ -304,6 +306,32 @@ class PackagesGetCommand extends FlutterCommand {
           environment: environment,
           buildSystem: globals.buildSystem,
         );
+      } else if (rootProject.directory.childFile('l10n.yaml').existsSync()) {
+        final Environment environment = Environment(
+          artifacts: globals.artifacts!,
+          logger: globals.logger,
+          cacheDir: globals.cache.getRoot(),
+          engineVersion: globals.flutterVersion.engineRevision,
+          fileSystem: globals.fs,
+          flutterRootDir: globals.fs.directory(Cache.flutterRoot),
+          outputDir: globals.fs.directory(getBuildDirectory()),
+          processManager: globals.processManager,
+          platform: globals.platform,
+          usage: globals.flutterUsage,
+          projectDir: rootProject.directory,
+          generateDartPluginRegistry: true,
+        );
+        final BuildResult result = await globals.buildSystem.build(
+          const GenerateLocalizationsTarget(),
```

Comment:
A few questions:

1. what's the difference between building the `GenerateLocalizationsTarget()` and calling `generateLocalizationsSyntheticPackage()`? From looking at the code it seems like the latter just runs a few extra validations, then calls the former.
2. Maybe I don't understand this synthetic package, but why would you want to opt out of the synthetic package and yet still have localizations generated?

Reply:
Synthetic package creates a "fake dart package" in `.dart_tool/flutter_gen` by adding the package to `.dart_tool/package_config.json`. But because of the way this is set up, it prevents users from exporting anything from the synthetic package (see https://github.com/flutter/flutter/issues/109606).

Reply:
So is the idea that I have:

1. flutter_app_a
2. flutter_app_b
3. localization_shared_package_c

The localizations all live in C, which A and B both depend on. So apps A and B both have `synthetic-package: false`, since they will be depending on real package C. But this PR will ensure running `flutter pub get` from A or B will still re-generate the localization files in C.

Is this right?

Reply:
I think that is the case, but why the `synthetic-package` flag is necessary (necessary because it prevents developers from exporting localization files) is separate from the purpose of this PR, which is to make sure that `gen_l10n` gets run in `flutter pub get` whether the `synthetic-package` flag is on or not.

Reply:
Right, but to understand the need for this change, I'm trying to understand why we even allow opting out of `synthetic-package`, since my understanding was that we relied on the synthetic package for localizations to work.

Reply:
Opting out of `synthetic-package` puts the generated files in the same project under `/lib/l10n`.

Reply:
I see, thanks